File: sdk/ruby/test/wire/workspaces_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class WorkspacesWireTest < WireMockTestCase
  def setup
    super

    @client = AgentTrunk::Client.new(
      access_token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_workspaces_list_with_wiremock
    test_id = "workspaces.list.0"

    @client.workspaces.list(request_options: {
      additional_headers: {
        "X-Test-Id" => "workspaces.list.0"
      }
    })

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/trunks",
      query_params: nil,
      expected: 1
    )
  end

  def test_workspaces_create_with_wiremock
    test_id = "workspaces.create.0"

    @client.workspaces.create(
      name: "name",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "workspaces.create.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/trunks",
      query_params: nil,
      expected: 1
    )
  end

  def test_workspaces_get_with_wiremock
    test_id = "workspaces.get.0"

    @client.workspaces.get(
      trunk_id: "trunkId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "workspaces.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/trunks/trunkId",
      query_params: nil,
      expected: 1
    )
  end

  def test_workspaces_audit_with_wiremock
    test_id = "workspaces.audit.0"

    @client.workspaces.audit(
      trunk_id: "trunkId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "workspaces.audit.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/trunks/trunkId/audit",
      query_params: nil,
      expected: 1
    )
  end
end

File: sdk/rust/tests/workspaces_test.rs
use agenttrunk::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_workspaces_list_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .workspaces
        .list(
            &WorkspacesListQueryRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/trunks", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_workspaces_create_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .workspaces
        .create(
            &CreateTrunkInput {
                name: "name".to_string(),
                description: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/trunks", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_workspaces_get_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client.workspaces.get(&"trunkId".to_string(), None).await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/trunks/trunkId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_workspaces_audit_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .workspaces
        .audit(
            &"trunkId".to_string(),
            &AuditQueryRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/trunks/trunkId/audit", None, 1)
        .await
        .unwrap();
}
